src/components/sections/ProductShelf/ProductShelf.tsx

```diff
@@ -1,34 +1,40 @@
 import React from 'react'
-import type { ProductSummary_ProductFragment } from '@generated/graphql'
 import ProductShelfSkeleton from 'src/components/skeletons/ProductShelfSkeleton'
+import { useProductsQuery } from 'src/sdk/product/useProductsQuery'
+import type { ProductsQueryQueryVariables } from '@generated/graphql'
 
 import ProductCard from '../../product/ProductCard'
 import Section from '../Section'
 
 import './product-shelf.scss'
 
-interface ProductShelfProps {
-  products: ProductSummary_ProductFragment[]
+interface ProductShelfProps extends ProductsQueryQueryVariables {
   title: string | JSX.Element
   withDivisor?: boolean
 }
 
 function ProductShelf({
-  products,
   title,
   withDivisor = false,
+  ...variables
 }: ProductShelfProps) {
+  const products = useProductsQuery(variables)
+
+  if (products?.edges.length === 0) {
+    return null
+  }
+
   return (
     <Section
       className={`layout__section ${withDivisor ? 'shelf__divisor' : ''}`}
     >
       <h2 className="text__title-section layout__content">{title}</h2>
       <div data-product-shelf-content>
-        <ProductShelfSkeleton loading={products.length === 0}>
+        <ProductShelfSkeleton loading={products == null}>
```

Comment:
```suggestion
        <ProductShelfSkeleton loading={products === null}>
```

Reply:
not really, we need to load on undefined too

Reply:
Use or operator, make the code more readable